File: app/src/main/java/com/popalay/yocard/ui/debts/DebtsFragment.java
package com.popalay.yocard.ui.debts;

import android.app.ActivityOptions;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.popalay.yocard.R;
import com.popalay.yocard.data.models.Debt;
import com.popalay.yocard.databinding.FragmentDebtsBinding;
import com.popalay.yocard.ui.adddebt.AddDebtActivity;
import com.popalay.yocard.ui.base.BaseFragment;
import com.popalay.yocard.utils.recycler.SimpleItemTouchHelperCallback;
import com.popalay.yocard.utils.transitions.FabTransform;

import java.util.List;

public class DebtsFragment extends BaseFragment implements DebtsView, SimpleItemTouchHelperCallback.SwipeCallback {

    @InjectPresenter DebtsPresenter presenter;

    private FragmentDebtsBinding b;
    private DebtsViewModel viewModel;
    
    public static DebtsFragment newInstance() {
        return new DebtsFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        b = DataBindingUtil.inflate(inflater, R.layout.fragment_debts, container, false);
        return b.getRoot();
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initUI();
    }

    @Override
    public void showAddDialog() {
        final Intent intent = AddDebtActivity.getIntent(getActivity());
        FabTransform.addExtras(intent, ContextCompat.getColor(getActivity(), R.color.accent), R.drawable.ic_write);
        ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(getActivity(), b.buttonWrite,
                getString(R.string.transition_add_debt));
        startActivity(AddDebtActivity.getIntent(getActivity()), options.toBundle());
    }

    @Override
    public void setItems(List<Debt> items) {
        viewModel.setDebts(items);
    }

    @Override
    public void showRemoveUndoAction(Debt item) {
        Snackbar.make(b.listDebts, R.string.debt_removed, Snackbar.LENGTH_SHORT)
                .setAction(R.string.action_undo, view -> presenter.onRemoveUndo(item))
                .show();
    }

    @Override
    public void onSwiped(RecyclerView.ViewHolder viewHolder) {
        final Debt debt = viewModel.get(viewHolder.getAdapterPosition());
        presenter.onItemSwiped(debt);
    }

    private void initUI() {
        viewModel = new DebtsViewModel();
        b.setModel(viewModel);
        
        b.buttonWrite.setOnClickListener(v -> presenter.onAddClick());
        new ItemTouchHelper(new SimpleItemTouchHelperCallback(this)).attachToRecyclerView(b.listDebts);
    }
}

File: presentation/src/main/kotlin/com/popalay/cardme/screens/debts/DebtsFragment.kt
package com.popalay.cardme.screens.debts

import android.app.ActivityOptions
import android.arch.lifecycle.ViewModelProvider
import android.content.Intent
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.popalay.cardme.R
import com.popalay.cardme.base.BaseFragment
import com.popalay.cardme.databinding.FragmentDebtsBinding
import com.popalay.cardme.utils.extensions.getDataBinding
import com.popalay.cardme.utils.extensions.getViewModel
import com.popalay.cardme.utils.extensions.unsafeActivity
import com.popalay.cardme.utils.extensions.unsafeContext
import com.popalay.cardme.utils.transitions.FabTransform
import javax.inject.Inject

class DebtsFragment : BaseFragment() {

    companion object {
        fun newInstance() = DebtsFragment()
    }

    @Inject lateinit var factory: ViewModelProvider.Factory

    private lateinit var b: FragmentDebtsBinding

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        b = getDataBinding<FragmentDebtsBinding>(inflater, R.layout.fragment_debts, container)
        b.vm = getViewModel<DebtsViewModel>(factory)
        return b.root
    }


    fun createAddDebtTransition(activityIntent: Intent): Bundle {
        FabTransform.addExtras(activityIntent, ContextCompat.getColor(unsafeContext, R.color.accent), R.drawable.ic_write)
        val options = ActivityOptions.makeSceneTransitionAnimation(unsafeActivity, b.buttonWrite, getString(R.string.transition_add_debt))
        return options.toBundle()
    }
}
